server/app.py

```diff
@@ -1,240 +1,184 @@
-def upload():
-    # save the game to the database
-    game = save_game(request.json)
-
-    # calculate and save stats
-    StatsCalculator(game).run()
-
-    # clear the stats cache
-    cache.clear()
+app = FastAPI(
+    title="OCUA Parity League",
+    summary="API Documentation",
+    openapi_tags=[
+        {"name": "api"},
+        {"name": "android"},
+        {"name": "admin"},
+    ],
+)
 
-    return ('', 201)
 
+# Assets
+react_app_path = Path(__file__).parents[1] / "web/build"
+if not react_app_path.exists():
+    print(f"Warning: React app directory not found at {react_app_path}")
 
-def save_game(upload_json):
-    game = Game()
 
-    game.league_id = upload_json['league_id']
-    game.week = upload_json['week']
+# Database Setup
+db_path = Path(__file__).parent / "db.sqlite"
+db_uri = "sqlite:////" + str(db_path.absolute())
+if os.name == "nt":
+    db_uri = "sqlite:///" + str(db_path.absolute())
 
-    game.home_team = upload_json['homeTeam']
-    game.away_team = upload_json['awayTeam']
+logging.basicConfig()
+logger = logging.getLogger("sqlalchemy.engine")
+logger.setLevel(logging.INFO)
 
-    game.home_score = upload_json['homeScore']
-    game.away_score = upload_json['awayScore']
+engine = create_engine(db_uri)
 
-    game.home_roster = upload_json['homeRoster']
-    game.away_roster = upload_json['awayRoster']
 
-    game.points = upload_json['points']
+# Database Session
+def get_session():
+    with Session(engine) as session:
+        yield session
 
-    db.session.add(game)
-    db.session.commit()
 
-    return game
+# Dependencies
+SessionDep = Annotated[Session, Depends(get_session)]
+AdminDep = Annotated[Session, Depends(admin.verify)]
 
 
-# API
-@cache.cached()
-@app.route('/api/<league_id>/teams')
-def teams(league_id):
-    teams = build_teams_response(league_id)
-    return jsonify(teams)
-
-
-@cache.cached()
-@app.route('/api/<league_id>/schedule')
-def schedule(league_id):
-    teams = build_teams_response(league_id)
-
-    matchup_count = len(teams) / 2
-    local_today = datetime.datetime.now() + datetime.timedelta(hours=league_utc_offset)
-    today = local_today.date()
+# Current League
+class League(api.BaseSchema):
+    id: int
+    name: str
+    line_size: int
 
-    query = Matchup.query.filter(Matchup.league_id == league_id, Matchup.game_start >= today).limit(matchup_count)
 
-    matchups = [matchup.to_dict() for matchup in query]
+class CurrentLeague(api.BaseSchema):
+    league: League
 
-    return jsonify({"teams": teams, "matchups": matchups})
 
+@app.get("/current_league", tags=["android"])
+async def current_league(session: SessionDep) -> CurrentLeague:
+    """Return the current league.
 
-@cache.cached()
-@app.route('/api/<league_id>/players')
-def players(league_id):
-    players = build_players_response(league_id)
-    return jsonify(players)
+    Used by the Android app which requires the nesting
+    """
+    league = session.get(db.League, CURRENT_LEAGUE_ID)
+    assert league
```

Comment:
is there a way to have sqlmodel raise something instead of doing this? (it is for mypy to not complain about an | None later).